packages/rollup-plugin/src/index.ts

```diff
@@ -104,5 +151,49 @@ export function vanillaExtractPlugin({
         map: null,
       };
     },
+
+    // Generate bundle (if extracting)
+    async buildEnd() {
+      if (!extract) {
+        return;
```

Comment:
In the 2 new build stages, we’re only running code if `extract` is specified, completing the loop on ensuring backwards-compatible behavior.